piccolo_api/crud/endpoints.py

```diff
@@ -196,6 +204,7 @@ def __init__(
         self.exclude_secrets = exclude_secrets
         self.validators = validators
         self.max_joins = max_joins
+        self.hooks = hooks if hooks else []
```

Comment:
One pattern we've used in the past is to create a dictionary, so we can look up things faster, rather than doing a list comprehension each time. For example:

```python
self.hook_map = {
    group[0]: [hook for hook in group[1]]
    for group in itertools.groupby(hooks, lambda x: x.hook_type)
}
```

The lookup is then just:

```python
self.hook_map.get(HookType.pre_save)
```

Instead of:

```python
[x for x in hooks if x.hook_type == hook_type]
```

The speed boost depends on the length of the list, but is generally worth it.

Reply:
thanks! So just so I understand you're suggesting to keep the list as the "public" api, but construct the hook_map attr during init(), and use that internally?

Reply:
Yes, that's right.